packages/cubejs-client-core/src/index.js

```diff
@@ -109,11 +109,17 @@ class CubejsApi {
       }
 
       let body = {};
-
+      let json = {};
+      let text = '';
+      // XXX: The cloned response stream and the original stream have to be
+      // resolved in parallel, otherwise if the response is bigger than the
+      // highWaterMark (16kB in node.js) the text stream will hang.
       try {
-        body = await response.clone().json();
+        const r1 = await response.clone();
+        ([json, text] = await Promise.all([response.json(), r1.text()]));
```

Comment:
This is not gonna work as if either of the promises fails `Promise.all` fails as well

Reply:
Hello @vasilev-alex,

Oops, you are of course right. I guess the `continue wait` I was testing this with does come inside a JSON response and does not exercise the catch path.

I guess we can avoid the clone altogether by reading the stream as text and then trying to parse the resulting text as JSON. If it fails, use the already read text for body.error?

I have amended by previous commit with this other approach. It still works for my test cases. What do you think?